ports/geckolib/gecko_bindings/sugar/refptr.rs

```diff
@@ -27,44 +37,62 @@ pub unsafe trait RefCounted where Self: Sized {
         };
 
         let result = cb(owned.as_ref());
+        // this isn't unwind safe, do not use without panic=abort
         forget(owned);
 
         result
     }
 
+    /// Given a non-null strong FFI reference, converts it into an Arc.
+    /// Panics on null.
     fn into(ptr: Strong<Self::FFIType>) -> Arc<Self> {
+        assert!(!ptr.is_null());
         unsafe { transmute(ptr) }
     }
 
+    /// Given a borrowed FFI reference, converts it to an Arc.
+    /// Unsafe to run on null values.
+    /// Can cause use-after-free if the Arc is allowed to have its destructor
+    /// run, please forget() arcs created by this
```

Comment:
That signature is unsafe because `&Arc` isn't safe to return for these types. You're describing `with()`.